packages/kit/src/exports/node/index.js

```diff
@@ -105,29 +105,36 @@ export async function getRequest({ request, base, bodySizeLimit }) {
 
 /** @type {import('@sveltejs/kit/node').setResponse} */
 export async function setResponse(res, response) {
-	const headers = Object.fromEntries(response.headers);
-
-	if (response.headers.has('set-cookie')) {
-		const header = /** @type {string} */ (response.headers.get('set-cookie'));
-		const split = set_cookie_parser.splitCookiesString(header);
-
-		// @ts-expect-error
-		headers['set-cookie'] = split;
+	for (const [key, value] of response.headers) {
+		try {
+			res.setHeader(
+				key,
+				key === 'set-cookie'
+					? set_cookie_parser.splitCookiesString(
+							// This is absurd but necessary, TODO: investigate why
+							/** @type {string}*/ (response.headers.get(key))
+					  )
+					: value
+			);
+		} catch (error) {
+			res.getHeaderNames().forEach((name) => res.removeHeader(name));
```

Comment:
Not sure on this one - should we just remove all headers, or just the ones up until the error, or try to parse the other headers and add all but the invalid ones? Are there any security implications to this?

Reply:
My thinking was that it's an entirely different response because of the error and it shouldn't carry over any of the original headers.

>  should we just remove all headers, or just the ones up until the error

In its current state at least, those two would mean the same as all the headers haven't yet been copied over to the `res` object yet, in essence we're undoing earlier iterations of this loop.